ccc/concourse.py

```diff
@@ -16,50 +16,95 @@
 
 import ci.util
 import concourse.client
+import concourse.client.api
 import ctx
+import model.concourse
 
-from ensure import ensure_annotations
+import ensure
 
 
-@functools.lru_cache()
-@ensure_annotations
-def client_from_cfg_name(
-    concourse_cfg_name: str,
-    team_name: str
-):
-    cfg_factory = ci.util.ctx().cfg_factory()
+def lookup_cc_team_cfg(cfg_set, team_name) -> model.concourse.ConcourseTeamConfig:
+    for cc_uam in cfg_set._cfg_elements('concourse_uam'):
+        if cc_uam.team_name() == team_name:
+            return cc_uam
 
-    cc_cfg = cfg_factory.concourse(concourse_cfg_name)
-    uam_cfg = cfg_factory.concourse_uam(cc_cfg.concourse_uam_config())
+    raise RuntimeError(f'No concourse uam for team name {team_name} found')
 
-    return concourse.client.from_cfg(
-        concourse_cfg=cc_cfg,
-        concourse_uam_cfg=uam_cfg,
+
+@ensure.ensure_annotations
+def client_from_parameters(
+    base_url: str,
+    password: str,
+    team_name: str,
+    username: str,
+    verify_ssl: bool = True,
+    concourse_api_version=None,
+) -> concourse.client.api.ConcourseApiBase:
+    """
+    returns a concourse-client w/ the credentials valid for the current execution environment.
+    The returned client is authorised to perform operations in the same concourse-team as the
+    credentials provided calling this function.
```

Comment:
add a brief comment explaining the difference / intention between `client_from_parameters` vs `client_from_cfg`.

also, on an additional note, how about merging them into one factory-function, as done in https://github.com/gardener/cc-utils/blob/master/ccc/clamav.py#L24 ?

team_name + cfg_set ==> no other arguments allowed
team_name + no cfg_set ==> require all other arguments